File: Chapter04/src/designpatterns/builder/java/CarBuilderExample.java
package designpatterns.builder.java;

public class CarBuilderExample {
    public static void main(String[] args) {
        Car car = new Car.Builder()
                .setName("Polo")
                .setColour("White")
                .setNumberOfDoors("4")
                .setTransmissionType("Automatic")
                .setFuelType("Petrol")
                .setCarType("Hatchback")
                .setEngine("1.2-litre TSI Petrol")
                .build();
    }
}

File: Chapter04/src/designpatterns/builder/kotlin/CarBuilderExample.kt
package designpatterns.builder.kotlin

fun main() {
    val car = CarV1(name = "Polo", colour = "White", breaks = "ABS", carType = "Hatchback")
    val petrolCar = CarV1(name = "Polo", colour = "White", breaks = "ABS", carType = "Hatchback", fuelType = "Petrol")
}